girder/models/user.py

```diff
@@ -425,18 +415,27 @@ def createUser(self, login, password, firstName, lastName, email,
 
         return user
 
-    def canLogin(self, user):
+    def canLogin(self, user, exc=False):
```

Comment:
IMO, the pattern of dynamically changing the return type / exception throwing based on arguments makes functions more difficult to understand and maintain.

I think a better pattern would be for this function to return a `(bool, str)`, optionally indicating a reason why login failed, but that would be a breaking change.

Perhaps an alternative is to implement a new `User` method with the desired behavior, then deprecate the old one.

Reply:
I agree with @brianhelba 

Union types or tuple return types are more familiar patterns.

Reply:
Third option is to duplicate the code that raises the exceptions in oauth plugin and don't even touch the girder core...